test/saithrift/tests/saiacl.py

```diff
@@ -1634,3 +1634,273 @@ def runTest(self):
+            assert acl_table_id > 0, 'acl_entry_id is <= 0'
+            assert acl_entry_id > 0, 'acl_entry_id is <= 0'
+            print '#### ACL \'DROP, src 12.12.12.1/255.255.255.0, in_ports[ptf_intf_2]\' Applied ####'
+
+            print "Sending packet from port 3 to port 1"
+            send_packet(self, 2, str(pkt))
+            print '#### Expecting 00:11:22:33:44:55 |', router_mac, '| 12.12.12.1 | 192.168.0.1 | @ ptf_intf 1'
+            verify_packets(self, exp_pkt, [0])
+
+            print "Sending packet from port 2 to port 1"
+            send_packet(self, 1, str(pkt))
+            print '#### NOT Expecting 00:11:22:33:44:55 |', router_mac, '| 12.12.12.1 | 192.168.0.1 | @ ptf_intf 1'
+            verify_no_packet(self, exp_pkt, 0)
+
+            print "Removing the ACL entry and table"
+            attr_value = sai_thrift_attribute_value_t(oid=SAI_NULL_OBJECT_ID)
+            attr = sai_thrift_attribute_t(id=SAI_PORT_ATTR_INGRESS_ACL, value=attr_value)
+            self.client.sai_thrift_set_port_attribute(port2, attr)
+
+            # unbind this ACL table from vlan object id
+            attr_value = sai_thrift_attribute_value_t(oid=SAI_NULL_OBJECT_ID)
+            attr = sai_thrift_attribute_t(id=SAI_VLAN_ATTR_INGRESS_ACL, value=attr_value)
+            self.client.sai_thrift_set_vlan_attribute(vlan_oid, attr)
+
+            print "Sending packet port 2 -> port 1 (192.168.0.1 -> 12.12.12.1 [id = 105])"
+            send_packet(self, 1, str(pkt))
+            print "Verifying packet"
+            verify_packets(self, exp_pkt, [0])
+
+            print "Sending packet port 3 -> port 1"
+            send_packet(self, 2, str(pkt))
+            print "Verifying packet at port 1"
+            verify_packets(self, exp_pkt, [0])
+
+        finally:
+            # unbind this ACL table from port and vlan object id
+            attr_value = sai_thrift_attribute_value_t(oid=SAI_NULL_OBJECT_ID)
+            attr = sai_thrift_attribute_t(id=SAI_PORT_ATTR_INGRESS_ACL, value=attr_value)
+            self.client.sai_thrift_set_port_attribute(port2, attr)
+
+            attr_value = sai_thrift_attribute_value_t(oid=SAI_NULL_OBJECT_ID)
+            attr = sai_thrift_attribute_t(id=SAI_VLAN_ATTR_INGRESS_ACL, value=attr_value)
+            self.client.sai_thrift_set_vlan_attribute(vlan_oid, attr)
+
+            sai_thrift_delete_fdb(self.client, vlan_oid, dmac2, port1)
+
+            # cleanup ACL
+            self.client.sai_thrift_remove_acl_entry(acl_entry_id)
+            self.client.sai_thrift_remove_acl_table(acl_table_id)
+            sai_thrift_remove_route(self.client, vr_id, addr_family, ip_addr3_subnet, ip_mask1, rif_id1)
+            self.client.sai_thrift_remove_next_hop(nhop1)
+            sai_thrift_remove_neighbor(self.client, addr_family, rif_id1, ip_addr1, dmac1)
+            sai_thrift_remove_lag_member(self.client, lag_member_id1)
+            self.client.sai_thrift_remove_lag(lag_id1)
```

Comment:
Need to remove rif_id1 before removing lag_id1, since rif_id1 is a port rif on lag_id1

Reply:
addressed comments